backend/app/core/database.py

```
@@ -62,16 +62,73 @@ def receive_after_cursor_execute(
+    for attempt in range(max_retries):
+        db = None
+        try:
+            db = get_fresh_db_session()
+            result = operation_func(db, *args, **kwargs)
+            return result
+        except Exception as e:
+            if db:
+                try:
+                    db.rollback()
+                except:
+                    pass
```

Ruff (0.12.2)

Use `contextlib.suppress(BaseException)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(BaseException)`

(SIM105)